apollo/src/graphql/storage-provider/storageProviderQueries.ts

```diff
@@ -27,4 +30,96 @@ builder.queryFields((t) => ({
             return storageProvider;
         },
     }),
+    listObjectsForModelVersion: t.field({
+        type: StorageProviderObjectConnection,
+        authScopes: {
+            loggedIn: true,
+        },
+        args: {
+            modelVersionId: t.arg.string({ required: true }),
+            // TODO: Putting defaultValue & required overrides defaultValue
+            first: t.arg({
+                type: 'Limit',
+                defaultValue: 10,
+                required: true,
+            }),
+            after: t.arg.string(),
+            prefix: t.arg.string(),
+        },
+        async resolve(_root, args, _ctx) {
+            const splitKey = (prefixLength: number, objectKey?: string): string[] => {
+                if (objectKey) {
+                    return objectKey.slice(prefixLength + 1).split('/');
+                }
+
+                return [];
+            };
+
+            const folderOrFile = (keyArr?: string[]): 'folder' | 'file' | undefined => {
+                if (keyArr) {
+                    return keyArr.length > 1 ? 'folder' : 'file';
+                }
+
+                return undefined;
+            };
+
+            const modelStorageProvider = (await ObjectionMLModelVersion.relatedQuery(
+                'storageProvider',
+            )
+                .for(args.modelVersionId)
+                .first()) as ObjectionStorageProvider;
+
+            const modelVersion = (await ObjectionMLModelVersion.query()
+                .findById(args.modelVersionId)
+                .first()) as ObjectionMLModelVersion;
+
+            // Appends new folder name to base prefix
+            const prefix = `${modelVersion.s3Prefix}`.concat(args.prefix ? '/' + args.prefix : '');
+
+            /* If no continuation token, the s3 api will always
+               return the root directory as an object. We do not want
+               this returned to the client. */
+            const limit = !args.after ? args.first + 1 : args.first;
+
+            // To get Pothos and the S3 API to play nicely
+            const maxKeys =
+                args.after === null || args.after === undefined ? undefined : args.after;
+
+            const { Contents, IsTruncated, NextContinuationToken, Prefix } = await ListObjects(
+                modelStorageProvider,
+                limit,
+                prefix,
+                maxKeys,
+            );
+
+            const prefixLength = prefix.length;
+
+            const objects = Contents
+                ? Contents.map((Content) => ({
+                      id: Content.Key,
+                      name:
+                          Content.Key &&
+                          folderOrFile(splitKey(prefix.length, Content.Key)) === 'folder'
```

Comment:
Why reference both `prefix.length` and the `prefixLength` variable?

Reply:
Cause I forgot to change it :)